branchpro/apps/_simulation.py

```diff
@@ -0,0 +1,121 @@
+#
+# SimulationApp
+#
+# This file is part of BRANCHPRO
+# (https://github.com/SABS-R3-Epidemiology/branchpro.git) which is released
+# under the BSD 3-clause license. See accompanying LICENSE.md for copyright
+# notice and full license details.
+#
+
+import pandas as pd
+import dash
+import dash_bootstrap_components as dbc
+
+import dash_core_components as dcc
+import dash_html_components as html
+
+import branchpro as bp
+
+
+class IncidenceNumberSimulationApp:
+    """IncidenceNumberSimulationApp Class:
+    Class for the dash app with figure and sliders.
+    """
+    def __init__(self):
+        self.app = dash.Dash(
+            __name__, external_stylesheets=[dbc.themes.BOOTSTRAP])
+        self.plot = bp.IncidenceNumberPlot()
+        self.sliders = bp._SliderComponent()
+
+        self.app.layout = dbc.Container(
+            [
+                html.H1('Example Title'),
+                dbc.Row(
+                    [
+                        dbc.Col(dcc.Graph(
+                            figure=self.plot.figure, id='myfig')),
+                        dbc.Col(self.sliders.get_sliders_div())
+                    ],
+                    align='center',
+                ),
+            ],
+            fluid=True,
+        )
+
+    def add_data(self, df):
+        """
+        Adds incidence data to the plot in the dash app.
+
+        Parameters
+        ----------
+        df
+            (pandas DataFrame) contains numbers of new cases by days.
+            Data stored in columns 'Time' and 'Incidence Number', respectively.
+        """
+        self.plot.add_data(df)
+
+    def add_simulator(self, simulator):
+        """
+        Simulates an instance of a model, adds it as a line to the plot and
+        adds sliders to the app.
+
+        Parameters
+        ----------
+        simulator
+            (SimulatorController) a BranchPro model and the time bounds
+            between which you run the simulator.
+        """
+        if not issubclass(type(simulator), bp.SimulationController):
+            raise TypeError('Simulatior needs to be a SimulationController')
+
+        model = simulator.model
+
+        if not issubclass(type(model), bp.BranchProModel):
+            raise TypeError('Models needs to be a BranchPro')
+
+        bounds = simulator.get_time_bounds()
+        mid_point = sum(bounds)/2
+
+        self.sliders.add_slider(
+            'Initial Cases', 'init_cond', 10.0, 0.0, 100.0, 10.0)
```

Comment:
Maybe we should set the slider starting conditions to be keyword arguments, with the default values as you have here.